samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -16,6 +16,106 @@ def define_for_evaled_arguments(&implementation)
   context.locals[var_atom]
 }
 
+# <+ ...>
 ZIL_BUILTINS[:+] = define_for_evaled_arguments { |arguments|
   arguments.inject(0, :+)
 }
+
+# <- ...>
+ZIL_BUILTINS[:-] = lambda { |arguments, context|
+  if arguments.length == 0
+    result = 0
+  elsif arguments.length == 1
+    result = 0 - eval_zil(arguments[0], context)
+  else
+    i = 1
+    result = eval_zil(arguments[0], context)
+    while i < arguments.length
+      result -= eval_zil(arguments[i], context)
+      i += 1
+    end
+  end
+
+  result
```

Comment:
the part except for no arguments can be expressed by
```rb
arguments.inject(:-)
```

which will just do
`(((a - b) - c) - d) - ...`
for a list `[a, b, c, d, ....]`


probably you can do the same with the other operators, too somehow ;)

Reply:
I made some updates to this.